monai/data/meta_tensor.py

```diff
@@ -140,6 +141,8 @@ def as_dict(self, key: str) -> dict:
     @property
     def affine(self) -> torch.Tensor:
         """Get the affine."""
+        if "affine" not in self.meta:
+            self.meta["affine"] = self.get_default_affine()
```

Comment:
this might mask potential bugs as our constructor ensures there is an affine. Maybe raise an exception?